### PR TITLE
Fixes #34632 - Add bookmarks to React content pages

### PR DESCRIPTION
**Pull request by chris1984**
#### What are the changes introduced in this pull request?

Tablewrapper now support bookmarks :beers: so adding them to the content tabs so we can make the new UI more 1:1 with the old one.

#### Considerations taken when implementing this change?

Traces and module_streams didn't work with bookmarks since who knows when. The bookmarks controller in Foreman looks at Angular for the controller name to gather the list of them. So I added the controller names to the traces and module_stream angular controllers.

I added unit tests as well for the React pages to test for the fake bookmark data that Jeremy created with the repo sets pr.

#### What are the testing steps for this pull request?

* Create some bookmarks and make sure they save on the following pages:
  * Errata
  * Traces
  * Module Streams
  * Packages
 
* Verify there are not an JS errors in the console
* Verify you can see the bookmarks after you save them in the dropdown list

### CODE REVIEW

**Comment by chris1984** [comment]
@lfu updated the controller and added some unit tests in that area to confirm the name

File: webpack/components/extensions/HostDetails/Tabs/PackagesTab/PackagesTab.js [hunk 1]
```diff
@@ -348,6 +348,7 @@ export const PackagesTab = () => {
           additionalListeners={[hostId, packageStatusSelected,
             activeSortDirection, activeSortColumn]}
           fetchItems={fetchItems}
+          bookmarkController="katello_erratum_packages"
```
Comment (lfu) [inline webpack/components/extensions/HostDetails/Tabs/PackagesTab/PackagesTab.js:351]: Is this not `katello_host_installed_packages`?

File: webpack/components/extensions/HostDetails/Tabs/__tests__/packagesTab.test.js [hunk 2]
```diff
@@ -16,6 +17,8 @@ const contentFacetAttributes = {
 };
 
 const hostname = 'test-host.example.com';
+const packageBookmarks = foremanApi.getApiUrl('/bookmarks?search=controller%3Dkatello_erratum_packages');
```
Comment (lfu) [inline webpack/components/extensions/HostDetails/Tabs/__tests__/packagesTab.test.js:20]: Same here.

File: engines/bastion_katello/test/content-hosts/content/content-host-packages-applicable.controller.test.js [hunk 3]
```diff
@@ -51,6 +51,11 @@ describe('Controller: ContentHostPackagesApplicableController', function() {
         expect($scope.table).toBeTruthy();
     });
 
+    it("verifies controller name", function() {
+        expect($scope.controllerName).toBeTruthy();
+        expect($scope.controllerName).toBe("katello_erratum_packages");
```
Comment (lfu) [inline engines/bastion_katello/test/content-hosts/content/content-host-packages-applicable.controller.test.js:56]: Is it the right controller?

**Review by jeremylenz — COMMENTED** [review]
Can you also check to see how difficult it'd be to fix https://projects.theforeman.org/issues/34659 here? I trust your judgment either way, just wanted to bring it to your attention

**Comment by Andrewgdewar** [comment]
> Can you also check to see how difficult it'd be to fix https://projects.theforeman.org/issues/34659 here? I trust your judgment either way, just wanted to bring it to your attention

@jeremylenz @chris1984 

I left a note on the mentioned issue above if you are interested further.

TLDR: 
Patternfly's iconography uses "solid" icons as standard and does not support "far" (outlined) out of the box, design approved the decision to be consistent to Patternfly's design vs Foreman's when the initial bookmark functionality was built.

File: webpack/components/extensions/HostDetails/Tabs/PackagesTab/PackagesTab.js (hunk 1 above)
Comment (chris1984) [reply webpack/components/extensions/HostDetails/Tabs/PackagesTab/PackagesTab.js:351]: Will change

File: webpack/components/extensions/HostDetails/Tabs/__tests__/packagesTab.test.js (hunk 2 above)
Comment (chris1984) [reply webpack/components/extensions/HostDetails/Tabs/__tests__/packagesTab.test.js:20]: Will fix

File: engines/bastion_katello/test/content-hosts/content/content-host-packages-applicable.controller.test.js (hunk 3 above)
Comment (chris1984) [reply engines/bastion_katello/test/content-hosts/content/content-host-packages-applicable.controller.test.js:56]: After our remote session, this is correct.

**Comment by jeremylenz** [comment]
Maybe Foreman should switch to use PF's icons then? Or should we just leave as-is?

let me know your thoughts. Seems that redmine issue will need updating either way.

**Comment by chris1984** [comment]
@lfu updated

**Review by lfu — APPROVED** [review]
LGTM 👍🏻
